sdjournal/journal.go

```diff
@@ -266,12 +549,17 @@ func (j *Journal) GetDataValue(field string) (string, error) {
 // turned off by setting it to 0, so that the library always returns the
 // complete data objects.
 func (j *Journal) SetDataThreshold(threshold uint64) error {
+	sd_journal_set_data_threshold, err := j.getFunction("sd_journal_set_data_threshold")
+	if err != nil {
+		return err
+	}
+
 	j.mu.Lock()
-	r := C.sd_journal_set_data_threshold(j.cjournal, C.size_t(threshold))
+	r := C.my_sd_journal_set_data_threshold(sd_journal_set_data_threshold, j.cjournal, C.size_t(threshold))
```

Comment:
It is not introduced by this PR, but there seems to be a type mismatch between `threshold uint64` and `C.size_t(threshold)`. I'm not sure what golang will do here, but I guess the threshold will be just truncated on 32b systems.